lib/http.go

```diff
@@ -0,0 +1,102 @@
+package lib
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/qri-io/qri/config"
+)
+
+const jsonMimeType = "application/json"
+
+// HTTPClient implements the qri http client
+type HTTPClient struct {
+	Addr string
+}
+
+// NewHTTPClient instantiates a new HTTPClient
+func NewHTTPClient(addr string) *HTTPClient {
+	return &HTTPClient{
+		Addr: addr,
+	}
+}
+
+// Call resolves the action by mapping it to the correct API endpoint and passes on parameters and context info
+// currently this is a fairly flat process but should evolve in the future
+func (c HTTPClient) Call(ctx context.Context, apiEndpoint config.APIEndpoint, methodOverride string, params interface{}, result interface{}) error {
+	// post should be the default method as we are almost always passing on a call from one instance to another
+	httpMethod := http.MethodPost
+	if methodOverride != "" {
+		httpMethod = methodOverride
+	}
+	// TODO(arqu): work out mimeType configuration/override per API endpoint
+	mimeType := jsonMimeType
+	addr := fmt.Sprintf("http://%s/%s", c.Addr, apiEndpoint)
```

Comment:
we don't always use `HTTP`. I think this means the `NewHTTPClient` constructor is going to need to be more robust.